src/paradigma/tremor/feature_extraction.py

```diff
@@ -0,0 +1,431 @@
+import numpy as np
+import pandas as pd
+from scipy import signal
+from paradigma.constants import DataColumns
+
+def compute_welch_periodogram(
+        values: list,
+        window_type: str = 'hann',
+        sampling_frequency: int = 100,
+        segment_length_s: float = 3,
+        overlap: float = 0.8,
+        spectral_resolution: float = 0.25
+    )-> tuple:
+    """Estimate power spectral density of the gyroscope signal using Welch's method.
+    
+    Parameters
+    ----------
+    values: List
+        The values of the signal (e.g., gyroscope data) of a single window.
+    window_type: str
+        The type of window to be used for the PSD (default: 'hann')
+    sampling_frequency: int
+        The sampling frequency of the signal (default: 100)
+    segment_length_s: float
+        The length of each segment in seconds (default: 3)
+    overlap: float
+        The overlap between segments as fraction (default: 0.8)
+    spectral_resolution: float
+        The spectral resolution of the PSD in Hz (default: 0.25)
+        
+    Returns
+    -------
+    tuple
+        Lists of PSD values and the corresponding frequencies
+    """
+
+    segment_length_n = sampling_frequency*segment_length_s
+    overlap_n = segment_length_n*overlap
+    window = signal.get_window(window_type, segment_length_n,fftbins=False)
+    nfft = sampling_frequency/spectral_resolution
+    
+    f, Pxx = signal.welch(values,sampling_frequency,window,segment_length_n,
```

Comment:
Voor dit soort functies is het best practice om eea uit te schrijven:

`signal.welch(
x=values,
fs=sampling_frequency,
...
)`

Dan (1) weet je zeker dat je variabelen mappen aan de juiste parameters, en (2) weten anderen beter wat er gebeurt.

Reply:
Aangepast!